### DISCUSSION TITLE
Looping to send email

### DISCUSSION THREAD

**Discussion by ducnhoxlatui** [Budibase/budibase#10902] [Help] [answered]
Hi guys!!!
I try to send email for any email which saved in row by looping in automation. I also use check IsSend if it sent or not. When i test, IsSend continue to count but mail can not be sent. I check the email step and found it "Blank". I dont know how and why. does any one meet this problem
![image](https://github.com/Budibase/budibase/assets/71976420/03068be7-be58-4281-bd5f-91b256219b3c)
![image](https://github.com/Budibase/budibase/assets/71976420/511d874b-e0ea-4127-8bd4-88af6885c433)
![image](https://github.com/Budibase/budibase/assets/71976420/0e1b077b-407c-452e-b9eb-adb9ba1a92e7)

Thanks u guy alot

**Comment by melohagan** [comment]
Hey @ducnhoxlatui 

Could you share a screenshot of your Send Email step please?

**Reply by ducnhoxlatui** [reply]
![image](https://github.com/Budibase/budibase/assets/71976420/9f01dc9c-092e-4ed6-90ee-86c878d20a69)
![image](https://github.com/Budibase/budibase/assets/71976420/3bf7677d-107d-4482-9cb2-0b20d0838ace)
 
This is my screenshot about that.

**Reply by melohagan** [reply]
Just as a test, can you replace the *HTML Contents* JavaScript binding value with some static text and see if the problem persists.

**Reply by ducnhoxlatui** [reply]
@melohagan I tried, but its still can not work.

**Comment by andz-bb** [comment, marked as answer]
Hi @ducnhoxlatui,

I've had a look at your app export and noticed that two loop steps were being applied to the Send Email step. It looks like there's a bug that allows multiple loops to be added to a single step. This causes the step to not loop over the provided input. I have logged it here: #10961

To get the step working again you just need to click the delete button on the Looping step until it disappears. When it does, you can then re-add the loop. It should now properly loop over the input you provide as expected. Hope this helps!

**Reply by melohagan** [reply]
Great debugging!

**Reply by ducnhoxlatui** [reply]
Hi @andz-bb ,
i have read the solution you share but can not understand your idea.
Beside that, i try to delete the second loop and it still can not send email.

**Reply by andz-bb** [reply]
@ducnhoxlatui for this issue I would delete the whole Send Email step and then add it again. This will make sure that there aren't any other loops causing problems on that step. 

Is the email problem the same as what you are mentioning in #10951?

**Reply by ducnhoxlatui** [reply]
in #10951, i have the solution that smtp i config need to be hosting by email such as student email or organization email because of the privacy of the hosting email not allow to host by gmail, yahoo or hot mail.
